### DISCUSSION TITLE
How can I fix the deprecated functions with M 2.4.4 and PHP 8.1

### DISCUSSION THREAD

**Discussion by krasho** [markshust/docker-magento#792] [General]
I am updating a project from M 2.4.0 to M 2.4.4 and I have seen some deprecated errors when I use PHP 8.1

for example this:
**Deprecated Functionality: Zend_Http_UserAgent implements the Serializable interface, which is deprecated.**

The file is:
vendor/magento/zendframework1/library/Zend/Http/UserAgent.php

**Class Zend_Http_UserAgent implements Serializable
{
}**

how can I fix this without changing the vendor file? I already saw messages to explode and strpos functions

thanks

**Comment by rangerz** [comment]
For not best practices...

Edit `app/boostrap.php` to add `error_reporting(E_ALL ^ E_DEPRECATED);`

**Reply by krasho** [reply]
thanks @rangerz 

I found this package in the composer.json file
 "magento/magento2-functional-testing-framework": "^3.7",

this package is adding the zendframework1 package.  

is it possible to remove that?

thanks

**Reply by rangerz** [reply]
If you test ok, it will be fine.

You may use `compose remove "spec module"` to remove, it will help you check the dependency.

**Reply by krasho** [reply]
thanks @rangerz 

no, this didn't work. I am trying your first recommendation.

but I have the same question, why is few people talking about this? doesn't this happen in a clean installation?

thanks
